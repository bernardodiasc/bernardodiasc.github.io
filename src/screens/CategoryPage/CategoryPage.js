import React, { Component } from 'react'

import Meta from 'components/Meta'
import LayoutContainer from 'containers/LayoutContainer'
import PageContent from 'displays/PageContent'
import CategoryDetails from 'displays/CategoryDetails'
import PostsListing from 'displays/PostsListing'

class CategoryPage extends Component {
  static defaultProps = {
    category: {},
    posts: [],
  }

  render() {
    return (
      <LayoutContainer>
        <Meta
          title={`Category: ${this.props.category.title}`}
          description={this.props.category.body}
        />
        <PageContent>
          <CategoryDetails category={this.props.category} />
          <PostsListing items={this.props.posts} />
        </PageContent>
      </LayoutContainer>
    )
  }
}

export default CategoryPage
