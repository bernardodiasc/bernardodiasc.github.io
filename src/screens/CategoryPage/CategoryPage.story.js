import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import StoryRouter from 'storybook-router'
import Provider from 'stories/containers/Provider'
import Docs from 'stories/containers/Docs'

import CategoryPage from './CategoryPage'
import CategoryPageReadme from './README.md'
storiesOf('Screens/CategoryPage', module)
  .addDecorator(story => <Provider story={story()} />)
  .addDecorator(StoryRouter())
  .addDecorator(withKnobs)
  .add('Info', () => <Docs>{CategoryPageReadme}</Docs>)
  .add('Example', () => <CategoryPage />)
  .add('Minimal render', () => <CategoryPage />)
