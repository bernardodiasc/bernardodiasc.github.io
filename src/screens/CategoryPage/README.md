# CategoryPage

## How to use

```
import CategoryPage from 'screens/CategoryPage'
```

```
<CategoryPage />
```

## Properties

| propName | propType | defaultValue | isRequired |
| - | - | - | - |
