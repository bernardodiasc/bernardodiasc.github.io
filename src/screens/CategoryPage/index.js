import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import {
  getAllCategoryDetails,
  getAllPostsForListingByCategory,
} from 'selectors/data'

import CategoryPage from './CategoryPage'

const mapStateToProps = (store, props) => ({
  category: getAllCategoryDetails(store, props.category),
  posts: getAllPostsForListingByCategory(store, props.category),
})

const mapDispatchToProps = dispatch => (
  bindActionCreators({
  }, dispatch)
)

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(CategoryPage)
