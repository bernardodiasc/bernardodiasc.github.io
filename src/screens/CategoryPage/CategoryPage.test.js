import React from 'react'
import { shallow } from 'enzyme'

import CategoryPage from './CategoryPage.js'

it('renders without crashing', () => {
  const component = shallow(<CategoryPage />)
  expect(component).toMatchSnapshot()
})
