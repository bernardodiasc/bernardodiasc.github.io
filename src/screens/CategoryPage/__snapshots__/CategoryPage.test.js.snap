// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Connect(LayoutContainer)>
  <Meta
    title="Category: undefined"
  />
  <PageContent>
    <CategoryDetails
      category={Object {}}
    />
    <PostsListing
      items={Array []}
    />
  </PageContent>
</Connect(LayoutContainer)>
`;
